auth/simple.go

```diff
@@ -70,3 +117,40 @@ func (a simpleAuth) GetUser(ctx context.Context) (string, error) {
 	}
 	return id.(string), nil
 }
+
+func (a simpleAuthN) authenticate(userID string) error {
+	if userID == "" {
+		return fmt.Errorf("Must provide header: [%s]", a.userIDHeader)
+	}
+	return nil
+}
+
+func (a simpleAuthZ) authorize(httpMethod, userID string) error {
+	switch httpMethod {
+	case "GET", "":
+		return a.authorizeUser(a.whitelistReads, a.readWhitelistedUserIDs, userID)
+	case "POST", "PUT", "PATCH", "DELETE":
+		return a.authorizeUser(a.whitelistWrites, a.writeWhitelistedUserIDs, userID)
+	default:
+		return nil
+	}
+}
+
+func (a simpleAuthZ) authorizeUser(useWhitelist bool, whitelistedUsers []string, userID string) error {
+	if !useWhitelist {
+		return nil
+	}
+
+	isAuthorized := false
+	for _, u := range whitelistedUsers {
+		if u == userID {
+			isAuthorized = true
+			break
+		}
+	}
+	if !isAuthorized {
```

Comment:
Can simply `return fmt.Errorf("supplied...")`, also note that error messages should start with lower-case letters.